## 1. Import the Pandas Library and Read the Dataset File in Spreadsheet Format based on the First - Fifth Debate Videos

### 1.1. DataFrame Variable from First Debate Video on KPU RI YouTube Channel 

In [1]:
"""
Melakukan proses import library Pandas serta memanggil file dataset dari hasil
web scraping komentar video YouTube debat capres - cawapres babak pertama di channel YouTube KPU RI
menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3 yang telah
disimpan atau di-export pada file berformat spreadsheet yang bernama "Export Comments Data from 1st Video Debate.xlsx"
"""

import pandas as pd

first_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 1st Video Debate.xlsx")

first_debate_video_df.head(3) # Menampilkan 3 data teratas yang dimiliki oleh variabel DataFrame "first_debate_video_df"

video_id                channel_id                  comment_id  \
0  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgzXkcSL0MuTJL6ZpY54AaABAg   
1  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgzF-687IdOdv5qeEsp4AaABAg   
2  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxbXCPeUDV_zPUhZjF4AaABAg   

             author                                       text_comment  \
0  @faridafasya9815  Kami rindu presiden seperti pak Anies semoga A...   
1    @rizkiindah479  Dulu dukung Prabowo Sandi, pasangan keren dan ...   
2  @nabilbudiman271  <a href="https://www.youtube.com/watch?v=yNO0Y...   

   like_count  total_reply                              channel_url  \
0         670           33  http://www.youtube.com/@faridafasya9815   
1         568           37    http://www.youtube.com/@rizkiindah479   
2         546           26  http://www.youtube.com/@nabilbudiman271   

       published_date        updated_date  
0 2023-12-12 15:24:47 2023-12-12 15:24:47  
1 2023-12-12 15:47:19 2023-12-12 15:47:19  
2 2023-12-12 20:40:59 2023-12-13 02:44:18

In [2]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "first_debate_video_df"
first_debate_video_df.shape

(5400, 10)

In [3]:
# Menampilkan 7 data terbawah yang dimiliki oleh variabel DataFrame "first_debate_video_df"
first_debate_video_df.tail(7)

video_id                channel_id                  comment_id  \
5393  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgwX9WcoRqi-SC3lCBx4AaABAg   
5394  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxUAGK3IbIDSSjvfjh4AaABAg   
5395  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgyK_w6cIUFhXPQF-9Z4AaABAg   
5396  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugy6Vqlcg3roF05-cyR4AaABAg   
5397  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxXcKhgFNQpHh56TUN4AaABAg   
5398  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgwLWqXkf_WLyCZAk7N4AaABAg   
5399  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugw2XXZdenwUrEezmEF4AaABAg   

                     author  \
5393           @rinideh6414   
5394     @althafkhaizan8299   
5395           @rinideh6414   
5396        @luthficell6059   
5397        @bdmchannel3564   
5398  @teresianurhayati6000   
5399  @ernamirzachannel8276   

                                           text_comment  like_count  \
5393  Anies tampil sangat meyakinkan, percaya diri d...           0   
5394  dengan mengamati hasil debat  capres,,saya kur...           0   
5395  memang dari debat ini terlihat sekali AMIN ada...           0   
5396  Menurut pandangan saya jika di lihat dari visi...           0   
5397  Semakin ragu dengan program kerjanya PRAGIB, m...           0   
5398                                    Ngga mudeng pak           0   
5399                                    bonyok prabowo😂           0   

      total_reply                                   channel_url  \
5393            0           http://www.youtube.com/@rinideh6414   
5394            0     http://www.youtube.com/@althafkhaizan8299   
5395            0           http://www.youtube.com/@rinideh6414   
5396            0        http://www.youtube.com/@luthficell6059   
5397            0        http://www.youtube.com/@bdmchannel3564   
5398            0  http://www.youtube.com/@teresianurhayati6000   
5399            0  http://www.youtube.com/@ernamirzachannel8276   

          published_date        updated_date  
5393 2023-12-13 11:01:07 2023-12-13 11:01:07  
5394 2023-12-13 11:00:12 2023-12-13 11:00:12  
5395 2023-12-13 10:59:02 2023-12-13 10:59:02  
5396 2023-12-13 10:58:16 2023-12-13 10:58:16  
5397 2023-12-13 10:57:19 2023-12-13 10:57:19  
5398 2023-12-13 10:55:40 2023-12-13 10:55:40  
5399 2023-12-13 03:10:40 2023-12-13 03:10:40

### 1.2. DataFrame Variable from Second Debate Video on KPU RI YouTube Channel 

In [4]:
"""
Melakukan proses pemanggilan (membaca) file dataset berformat Microsoft Excel (spreadsheet) dari hasil
web scraping komentar video YouTube debat capres - cawapres babak kedua di channel YouTube KPU RI
menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3
"""

second_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 2nd Video Debate.xlsx")

second_debate_video_df.head(11) # Menampilkan 11 data teratas yang dimiliki oleh variabel DataFrame "second_debate_video_df"

video_id                channel_id                  comment_id  \
0   YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  UgyBL7L6U5bXWuuzlap4AaABAg   
1   YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  Ugx-_DJpJiZRtfv7FMF4AaABAg   
2   YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  UgyTWKUniSQGlIiAAdd4AaABAg   
3   YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  UgxJyLPfPdq4lY8kmM14AaABAg   
4   YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  UgzV18Q57xtd1Q5EhGJ4AaABAg   
5   YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz8beukcta7PXEKGzZ4AaABAg   
6   YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  UgwZpYDNegbybA9EHdp4AaABAg   
7   YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  UgyzHeGaUMUoE2PGUa14AaABAg   
8   YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  UgzdD672bUnmr4eH6014AaABAg   
9   YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  UgyhsMzF4OvU-E9ykB94AaABAg   
10  YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  UgydOoeSIqVBdg4bo2Z4AaABAg   

                    author                                       text_comment  \
0         @nabilbudiman271  <a href="https://www.youtube.com/watch?v=YzC82...   
1    @aniksetianingsih2011  Pak gibran, saya sangat mendukung program kerj...   
2           @staypolid5125  GW SEBAGAI GENERASI MUDA SANGAT MENDUKUNG PENU...   
3               @bangjok22  wow,,, mas gibran,, awalnya gue ragu  mau pili...   
4            @fakihbae8975  Kok debat cawapres kayaknya lebih prepare dri ...   
5       @ronikurniawan8593  Masyaallah tabarakallah, Mas Gibran. Diawal sa...   
6           @hanifapau8563  Kolaborasi yg sempurna Prabowo dan Gibran, Pej...   
7          @tomieverywhere  Sehat2 slalu mas gibran, wlwpun terlihat kuran...   
8   @pinkblossomsshardynti  Salut sama mas gibran, pembicaraannya lugas da...   
9         @user-bt9bk1bn4l  Gibran diluar ekspektasi gokil si belimbing sa...   
10             @gegeyb9432  Awalnya ragu sama mas Gibran...ternyata cuma s...   

    like_count  total_reply                                    channel_url  \
0          750           73        http://www.youtube.com/@nabilbudiman271   
1          294           45   http://www.youtube.com/@aniksetianingsih2011   
2          289           21          http://www.youtube.com/@staypolid5125   
3          232           17              http://www.youtube.com/@bangjok22   
4          195           28           http://www.youtube.com/@fakihbae8975   
5          172           13      http://www.youtube.com/@ronikurniawan8593   
6          165            8          http://www.youtube.com/@hanifapau8563   
7          156            6         http://www.youtube.com/@tomieverywhere   
8          154            5  http://www.youtube.com/@pinkblossomsshardynti   
9          150           11        http://www.youtube.com/@user-bt9bk1bn4l   
10         132           19             http://www.youtube.com/@gegeyb9432   

        published_date        updated_date  
0  2023-12-22 15:35:34 2023-12-23 02:18:45  
1  2023-12-22 15:54:27 2023-12-22 15:54:27  
2  2023-12-22 15:32:16 2023-12-22 15:32:16  
3  2023-12-22 17:26:54 2023-12-22 17:26:54  
4  2023-12-22 14:53:57 2023-12-22 14:53:57  
5  2023-12-22 16:04:19 2023-12-22 16:04:19  
6  2023-12-22 18:04:44 2023-12-22 18:04:44  
7  2023-12-22 15:28:34 2023-12-22 15:28:34  
8  2023-12-22 17:08:09 2023-12-22 17:08:09  
9  2023-12-22 15:58:55 2023-12-22 15:58:55  
10 2023-12-22 17:00:27 2023-12-22 17:00:27

In [5]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "second_debate_video_df"
second_debate_video_df.shape

(1900, 10)

### 1.3. DataFrame Variable from Third Debate Video on KPU RI YouTube Channel

In [6]:
"""
Melakukan proses pemanggilan (membaca) file dataset berformat
Microsoft Excel (spreadsheet) dari hasil web scraping komentar video YouTube
debat capres - cawapres babak ketiga di channel YouTube KPU RI
yang menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3
"""

third_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 3rd Video Debate.xlsx")

third_debate_video_df.head(9)  # Menampilkan 9 data teratas yang dimiliki oleh variabel DataFrame "third_debate_video_df"

video_id                channel_id                  comment_id  \
0  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugy7bfVbuDxBgae9eVp4AaABAg   
1  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgwXBn-0QCB6GntoHY94AaABAg   
2  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugw-FFWX0IY1b5sW77x4AaABAg   
3  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgyLvJxPNZznf4VD_hF4AaABAg   
4  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugxo25xaQwZnNIDfCkl4AaABAg   
5  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgxEU6Pyh8OU1kV_am54AaABAg   
6  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgxZ7WP3Np2jAOVGaOl4AaABAg   
7  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgyUHi1eyviFsjfdXtl4AaABAg   
8  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz0_cQ4OgpUqNaLtzh4AaABAg   

                    author                                       text_comment  \
0         @nabilbudiman271  <a href="https://www.youtube.com/watch?v=KJdt-...   
1           @Teman_Bambank  Mari kita lihat sampai akhir, apakah KPU benar...   
2             @waryati5554  Saya sebagai org awam pun sangat mengerti kena...   
3    @angkadandata_channel  Smg presiden yg terpilih nantinya bisa memimpi...   
4  @ismailash-shanhaji1053  Saya tidak butuh pemimpin yang pandai Kata-kat...   
5          @vidiokeren6134                Anies berani banget. Saya yg takut.   
6      @wahyuuusaputra3524  SEMOGA NAJWA SIHAB DAPAT MEMPERTEMUKAN MEREKA ...   
7           @rialfachrezi8  siapapun presidennya saya harap kedepannya bis...   
8           @diary_nay7617  Pak prabowo seorang yg profesional tdk akan ba...   

   like_count  total_reply                                     channel_url  \
0        1590          122         http://www.youtube.com/@nabilbudiman271   
1         631           66           http://www.youtube.com/@Teman_Bambank   
2         432           90             http://www.youtube.com/@waryati5554   
3         384           47    http://www.youtube.com/@angkadandata_channel   
4         256           41  http://www.youtube.com/@ismailash-shanhaji1053   
5         229           43          http://www.youtube.com/@vidiokeren6134   
6         222           10      http://www.youtube.com/@wahyuuusaputra3524   
7         221           14           http://www.youtube.com/@rialfachrezi8   
8         212           25           http://www.youtube.com/@diary_nay7617   

       published_date        updated_date  
0 2024-01-07 15:37:05 2024-01-07 15:37:05  
1 2024-01-07 15:20:09 2024-01-07 15:20:09  
2 2024-01-08 03:40:07 2024-01-08 03:40:07  
3 2024-01-08 08:12:00 2024-01-08 08:12:00  
4 2024-01-07 19:06:31 2024-01-07 19:06:31  
5 2024-01-07 15:59:58 2024-01-07 15:59:58  
6 2024-01-07 17:09:06 2024-01-07 17:09:06  
7 2024-01-07 14:52:18 2024-01-07 14:53:56  
8 2024-01-07 17:18:16 2024-01-07 17:18:16

In [7]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "third_debate_video_df"
third_debate_video_df.shape

(2600, 10)

### 1.4. DataFrame Variable from Fourth Debate Video on KPU RI YouTube Channel

In [8]:
"""
Melakukan proses pemanggilan/membaca file dataset yang disimpan pada Microsoft Excel (spreadsheet) dari hasil
web scraping komentar video YouTube debat capres - cawapres babak keempat di channel YouTube KPU RI
yang menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3
"""

fourth_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 4th Video Debate.xlsx")

fourth_debate_video_df.head(17) # Menampilkan 17 data teratas yang dimiliki oleh variabel DataFrame "fourth_debate_video_df"

video_id                channel_id                  comment_id  \
0   anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  Ugy8BsMRwY9YzhEKdUN4AaABAg   
1   anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  UgwqbwQEnoha-UOrRjB4AaABAg   
2   anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  UgzIzxXzSws_MmRDt2h4AaABAg   
3   anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  UgxCMcgii0hrW8Mcfe14AaABAg   
4   anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  UgyPNxDdA5XXS4wQyZh4AaABAg   
5   anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  UgxS0ag0lJdVJHEXsIx4AaABAg   
6   anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  UgxpUqp8HQIQ-AXcN4V4AaABAg   
7   anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  UgyHVlj_JoWTVcxegvN4AaABAg   
8   anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  UgxYeQq7O1_zETn1VOx4AaABAg   
9   anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  UgwGBIBrJ_uCxCU01kt4AaABAg   
10  anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  UgxgJmuYpgDmjWkrPQh4AaABAg   
11  anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  UgyGczO3l5p3bbw4EP54AaABAg   
12  anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  Ugwm3gJ81UQiwd8ezop4AaABAg   
13  anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz1X-ixgzb0UR1sYL54AaABAg   
14  anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz_xrCeyNYh1X_a-D14AaABAg   
15  anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  Ugx-NitF_XNyymH3n2x4AaABAg   
16  anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  UgwcBBLtt8cqwxR4V-F4AaABAg   

                         author  \
0                  @kajikampung   
1           @maulidarizqika4610   
2                   @tfkqrkhman   
3                   @desybs6832   
4             @bobhaswendra3926   
5          @indrafahmiavery8212   
6                @diahjunia9464   
7            @dedi.official_Ns.   
8              @SeyChanel-yu6rh   
9                   @gordin1996   
10                 @yudosri6375   
11                @muflihrz4074   
12   @budiharyantokurniawan3803   
13             @user-hi1hs3qq6j   
14            @deafinantaka7022   
15  @virgiawanvirlislimanto1305   
16             @dodisaputra7814   

                                         text_comment  like_count  \
0   Saya sangat suka dengan para pengkritik yg int...         294   
1   Terimakasih Pak Mahfud dan Cak Imin, sangat ba...         190   
2   Pak Muhaimin ngmongnya daging semua, ada masal...         190   
3   Pak Mahfud sgt baik memaparkan data dan fakta....         129   
4   KPU, Tolong untuk debat terakhir, gak perlu la...         121   
5   Pengalaman dan jam terbang ga bisa bohong.. Pa...         118   
6   Prof Mahmud sdh jelas &amp; pasti kwalitasnya....         117   
7   Di sini bisa kita belajar siapa yang paling be...         112   
8   Sya Tertarik dengan Penyampaian Mahfud MD, ilm...         109   
9   Bapak mahfud benar kami masyarakat NTT, Flores...         109   
10  Semoga Alloh swt.meridzoi paslon&quot;yg Amanah..         101   
11  saya bangga sama pak Mahfud dan Cak Imin<br>Kh...         101   
12  semakin yakin gk akan pilih gibran,. dah itu a...          94   
13  Semoga siapa yg terpilih menjadi wakil rakyat,...          94   
14        Cak imin paling terarah penjelasan nya....❤          89   
15             Keren Mahfud MD debat malam ini 🤟🏼🤟🏼🤟🏼          76   
16  Waduh berat menyaingi bapak Mahfud.kapasitas.k...          74   

    total_reply                                        channel_url  \
0            19                http://www.youtube.com/@kajikampung   
1             8         http://www.youtube.com/@maulidarizqika4610   
2            22                 http://www.youtube.com/@tfkqrkhman   
3             6                 http://www.youtube.com/@desybs6832   
4             5           http://www.youtube.com/@bobhaswendra3926   
5             3        http://www.youtube.com/@indrafahmiavery8212   
6             7              http://www.youtube.com/@diahjunia9464   
7            18          http://www.youtube.com/@dedi.official_Ns.   
8             0            http://www.youtube.com/@SeyChanel-yu6rh   
9            12                 http://www.youtube.com/@gordin1996   
10            7                http://www.youtub

In [9]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "fourth_debate_video_df"
fourth_debate_video_df.shape

(1040, 10)

### 1.5. DataFrame Variable from Fifth Debate Video on KPU RI YouTube Channel

In [10]:
"""
Melakukan proses pemanggilan/membaca file dataset yang disimpan pada Microsoft Excel (spreadsheet) dari hasil
web scraping komentar video YouTube debat capres - cawapres babak kelima di channel YouTube KPU RI
yang menggunakan API dari salah satu service Google Client yaitu YouTube Data API v3
"""

fifth_debate_video_df = pd.read_excel("Results of Data Exports from DataFrame/Export Comments Data from 5th Video Debate.xlsx")

fifth_debate_video_df.head(12) # Menampilkan 12 data teratas yang dimiliki oleh variabel DataFrame "fifth_debate_video_df"

video_id                channel_id                  comment_id  \
0   8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz8GZ3Lfw7X9kYnqSV4AaABAg   
1   8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgwOYSRHpyihNxbUL_l4AaABAg   
2   8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgzE0Z69Qxhmo7W7rG54AaABAg   
3   8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgzZITGXd0oboGAaMEp4AaABAg   
4   8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz2mQvTxHONMD982514AaABAg   
5   8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgwHox027DBKmVb_9e54AaABAg   
6   8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugx8Uk2IjomoD16znX94AaABAg   
7   8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugy-S-3oeLCUjmfVhUt4AaABAg   
8   8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgyrHNJc5P3tu7NFebd4AaABAg   
9   8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgzO353BZYwH1CngstB4AaABAg   
10  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugxs0hfSDxhmzmixLcR4AaABAg   
11  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgyTOOItpkpJfObu5ll4AaABAg   

                        author  \
0           @anisluthfiani9056   
1              @sodrishandy545   
2                 @robinong153   
3                 @bintang4783   
4               @misyebima8609   
5                     @sxn1707   
6             @Nurwahida_Ahmad   
7          @nurulmutmainna1914   
8         @maslahatilummah2043   
9            @muhammadsamir510   
10  @srimurdiyatimurdiyati8326   
11             @elfisyarif2671   

                                         text_comment  like_count  \
0   Abah terimakasih sudah mewarnai demokrasi ini ...         885   
1   Semoga Allah hadirkan pemimpin yang adil bagi ...         376   
2   Terus terang..debat hari ini lebih berkualitas...         326   
3   Kmi berharap yg paling dalam, KPU dan bawaslu ...         274   
4   Ya Allah menyentuh banget penyampaian terakhir...         271   
5   AMIN sak jossse gak ada keraguan lagi...Bismillah         271   
6   Abah, kami optimis tapi ditangan Allah semua k...         260   
7   Abahhh apapun hasilnya nanti kami bangga pada ...         233   
8   Saya menjadi saksi di debat pertama hingga ter...         187   
9   JakLinko,Transjakarta,Street Complete,Trotoar,...         183   
10  Abah… terima kasih .. sdh menyampaikan yg terb...         178   
11           Gerakan Indonesia Bersama Anies Baswedan         173   

    total_reply                                        channel_url  \
0           227          http://www.youtube.com/@anisluthfiani9056   
1            10             http://www.youtube.com/@sodrishandy545   
2            20                http://www.youtube.com/@robinong153   
3            16                http://www.youtube.com/@bintang4783   
4            26              http://www.youtube.com/@misyebima8609   
5            22                    http://www.youtube.com/@sxn1707   
6            22            http://www.youtube.com/@Nurwahida_Ahmad   
7            32         http://www.youtube.com/@nurulmutmainna1914   
8            10        http://www.youtube.com/@maslahatilummah2043   
9            30           http://www.youtube.com/@muhammadsamir510   
10           16  http://www.youtube.com/@srimurdiyatimurdiyati8326   
11           15             http://www.youtube.com/@elfisyarif2671   

        published_date        updated_date  
0  2024-02-04 16:12:53 2024-02-04 16:12:53  
1  2024-02-04 15:27:34 2024-02-04 15:27:34  
2  2024-02-04 16:34:43 2024-02-04 16:34:43  
3  2024-02-04 17:17:52 2024-02-04 17:43:07  
4  2024-02-05 05:14:30 2024-02-05 05:14:30  
5  2024-02-04 16:11:05 2024-02-04 16:11:05  
6  2024-02-04 15:51:08 2024-02-04 15:51:08  
7  2024-02-04 15:11:35 2024-02-04 15:11:35  
8  2024-02-04 17:55:15 2024-02-04 17:55:15  
9  2024-02-04 15:47:40 2024-02-04 15:47:40  
10 2024-02-05 07:18:47 2024-02-05 07:18:47  
11 2024-02-04 15:13:17 2024-02-04 15:13:17

In [11]:
# Menampilkan jumlah baris dan kolom yang dimiliki oleh variabel DataFrame "fifth_debate_video_df"
fifth_debate_video_df.shape

(1800, 10)

In [12]:
# Menampilkan 8 data terbawah yang dimiliki oleh variabel DataFrame "fifth_debate_video_df"
fifth_debate_video_df.tail(8)

video_id                channel_id                  comment_id  \
1792  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgzhoddgrUPP83dT0t54AaABAg   
1793  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz2hYFmk7PWa7BQoXV4AaABAg   
1794  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgyQ2MRlfYE1vxyBu3p4AaABAg   
1795  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgxHk2YguNKSdcx_Mcp4AaABAg   
1796  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgxV7GJULI9y3HvArNF4AaABAg   
1797  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgzMxktSEwtzZfG2Fa54AaABAg   
1798  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugx_JIr_4N5AC8KJcbV4AaABAg   
1799  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugx4nJkrQUve_UheMdp4AaABAg   

                      author  \
1792              @Lilis-279   
1793     @dudehdewangsih8541   
1794  @sitinadasalsabila9520   
1795  @sitinadasalsabila9520   
1796       @tonykustiadi3213   
1797   @syahrulramadhani2365   
1798        @rasmibaderi1057   
1799            @opanboc5038   

                                           text_comment  like_count  \
1792  KPU yang terhormat tolong awasi dan pantau sel...           0   
1793  Siapapun president nya ketika dia amanah jujur...           0   
1794                                       Amin Aamiin.           0   
1795                                                😮😮😮           0   
1796             TUHAN YME YG MENENTUKAN PEMIMPIN NKRI❤           0   
1797  l😊😊lp<br>p.o..olloo.😊.ok<br>o<br>lloo😊ol.ooo😊o...           0   
1798  Yg lebih penting juga, masalah pendidikan kepe...           0   
1799  Gak ada Keraguan Untuk Pasangan AMIN....Seanda...           0   

      total_reply                                    channel_url  \
1792            0              http://www.youtube.com/@Lilis-279   
1793            0     http://www.youtube.com/@dudehdewangsih8541   
1794            0  http://www.youtube.com/@sitinadasalsabila9520   
1795            0  http://www.youtube.com/@sitinadasalsabila9520   
1796            0       http://www.youtube.com/@tonykustiadi3213   
1797            0   http://www.youtube.com/@syahrulramadhani2365   
1798            0        http://www.youtube.com/@rasmibaderi1057   
1799            0            http://www.youtube.com/@opanboc5038   

          published_date        updated_date  
1792 2024-02-04 22:37:14 2024-02-04 22:37:14  
1793 2024-02-04 22:38:29 2024-02-04 22:38:29  
1794 2024-02-04 22:41:39 2024-02-04 22:41:39  
1795 2024-02-04 22:42:06 2024-02-04 22:42:06  
1796 2024-02-04 22:45:54 2024-02-04 22:45:54  
1797 2024-02-04 22:50:55 2024-02-04 22:50:55  
1798 2024-02-04 23:01:15 2024-02-04 23:01:15  
1799 2024-02-04 18:22:22 2024-02-04 18:22:22

## 2. Combining Five DataFrame Variables Contain Comments Information from the First - Fifth Debate Videos on KPU RI YouTube Channel

In [13]:
"""
Menggabungkan 5 variabel DataFrame menjadi 1 file dataset dalam format
spreadsheet Microsoft Excel dimana masing-masing variabel DataFrame memiliki jumlah
data yang disimpan berbeda-beda karena saat jumlah data yang didefinisikan saat
melakukan proses scraping/pengambilan informasi-informasi komentar utama dengan metode pengulangan (looping)
dibuat berbeda antara jumlah data komentar dari video debat pertama, debat kedua, dst
"""

df_comment_video_debate_kpuri = pd.concat([first_debate_video_df, second_debate_video_df, third_debate_video_df,
                                           fourth_debate_video_df, fifth_debate_video_df],
                                          ignore_index = True)

# Menampilkan 25 data teratas yang dimiliki oleh variabel DataFrame "df_comment_video_debate_kpuri"
df_comment_video_debate_kpuri.head(25)

video_id                channel_id                  comment_id  \
0   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgzXkcSL0MuTJL6ZpY54AaABAg   
1   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgzF-687IdOdv5qeEsp4AaABAg   
2   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxbXCPeUDV_zPUhZjF4AaABAg   
3   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugw_i1DjvR-QcFZOWFR4AaABAg   
4   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxZddA-eb4ajHIz-vd4AaABAg   
5   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxCVWBXBHS2aAExXJ14AaABAg   
6   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugx2150EevOnGTsC6bB4AaABAg   
7   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgwG8FVmd_-g6-XkTVd4AaABAg   
8   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxJgM761WF2PFOlnoJ4AaABAg   
9   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgwiUVXS_YjXcTJ0ULB4AaABAg   
10  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgwBmgAzm9kt0Uu1_Et4AaABAg   
11  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugxm0_t-p32oVgxoVn54AaABAg   
12  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgyNp2gFN-BOUgKJlgB4AaABAg   
13  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgwFl1-jiOxcmyWJN-94AaABAg   
14  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxCr7dsBpQMoZwK-A14AaABAg   
15  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgyNSAHJafH7-F_rTXF4AaABAg   
16  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugy_U62VFyEpcKXofLp4AaABAg   
17  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugyp19hoc3VcL1rB-E94AaABAg   
18  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxYqA9ZtLRTz4fsp-14AaABAg   
19  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxsXJ5JNn84z3hZ_nR4AaABAg   
20  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugy7oxcK9nLg1UQjfnp4AaABAg   
21  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz4JfhGSm1UAI7NJmV4AaABAg   
22  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgzLDjz6-AjqoDnRWsJ4AaABAg   
23  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgwdWA1N0WtPQ0nhW9x4AaABAg   
24  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxpQ5BitRNKNhsLg894AaABAg   

                     author  \
0          @faridafasya9815   
1            @rizkiindah479   
2          @nabilbudiman271   
3       @ajipratamaputra815   
4       @ariesty_crispy2468   
5             @viedaily3572   
6                 @isalutfi   
7   @afiyahsetianingrum0158   
8            @indahraya4390   
9        @hendriakhtiri5839   
10             @herisalmani   
11               @restimita   
12           @dzikrifeature   
13            @muflihrz4074   
14    @inspirasichannel1570   
15     @prayogatribagus1676   
16           @ekamustika497   
17          @Awan_Saputra98   
18         @MasaDepan-lf3vu   
19      @AnakPantaiBali2024   
20      @engkushermawan2400   
21          @ahmadahmad8609   
22            @ujanguje4594   
23       @fatkhurrohman2889   
24             @bintang9593   

                                         text_comment  like_count  \
0   Kami rindu presiden seperti pak Anies semoga A...         670   
1   Dulu dukung Prabowo Sandi, pasangan keren dan ...         568   
2   <a href="https://www.youtube.com/watch?v=yNO0Y...         546   
3   Ingat kata Rocky Gerung<br>Argumen harus di ja...         403   
4   Mudha2n tidak ada kecurangan di pemilu nanti s...         287   
5          YUK KAWAL KPU RI AGAR TIDAK ADA KECURANGAN         267   
6   NO 1 PAK ANIES PRESIDEN. AMIN. ARGUMENTASINYA ...         244   
7   Yang mampu menyelamatkan indonesia hanya denga...         238   
8   Bismillah, dukung Pak Anis Rasyid Baswedan dan...         234   
9   Menurut gw untuk debat kali ini pak anies mena...         233   
10  Cuma bisa berharap KPU tetaplah menjadi wasit ...         232   
11  Terkagum kagum dengan pak anis , sehat selalu ...         231   
12  Semoga KPU bisa netral dan jujur supaya mengha...         218   
13  hasil debat hari ini<br>Anies: Memperbaiki Ind...         213   
14  Pak Anies secara penyampaian Visi dan Misi san...         203   
15  Disini sudah jelas terlihat mana yg sanggup me...         202   
16  Insyaallah Pak Anies untuk Indonesia yang lebi...         194   
17  harapan kami KPU tetap adil,netral dan  transp...      

Berdasarkan code diatas, saya telah berhasil menggabungkan 5 variabel DataFrame yang masing-masing DataFrame memiliki data/informasi komentarnya masing-masing. Sebagai contoh pada variabel DataFrame "first_debate_video_df" berisikan 5.400 data/informasi komentar yang berhasil diambil dari video debat babak pertama capres-cawapres Republik Indonesia tahun 2024 yang ditayangkan oleh YouTube channel KPU RI. Sedangkan variabel DataFrame "second_debate_video_df" juga memiliki data/informasi komentar utama namun yang membedakannya dengan DataFrame "first_debate_video_df" adalah jumlah data yang diambil dan sumber videonya karena jika variabel DataFrame "second_debate_video_df" memiliki 1.900 dan sumber informasi/data komentar yang diambil berasal dari video debat babak kedua capres-cawapres Republik Indonesia tahun 2024 yang di-upload oleh YouTube channel KPU RI.

Hal tersebut juga sama dengan yang dimiliki oleh variabel DataFrame lainnya seperti "third_debate_video_debate_df", "fourth_debate_video_df" dan "fifth_debate_video_df" dengan catatan jumlah data yang diambil berbeda-beda dan sumber video atas informasi komentar yang diambil juga berbeda namun video-video tersebut tetap di-upload/ditayangkan oleh YouTube channel KPU RI.

Untuk mengecek dan memastikan apakah penggabungan 5 variabel DataFrame berhasil dilakukan, saya akan menampilkan 6 data terbawah yang dimiliki oleh DataFrame "df_comment_video_debate_kpuri" karena jika proses penggabungan 5 variabel DataFrame menjadi 1 DataFrame berhasil dilakukan maka 6 data terbawah seharusnya berasal dari data-data komentar yang diambil (di-scraping) dari video debat babak kelima capres-cawapres Republik Indonesia tahun 2024 yang ditayangkan oleh YouTube channel KPU RI dan disimpan di dalam variabel DataFrame bernama "fifth_debate_video_df".

Sebagai penjelasan tambahan saat saya melakukan proses penggabungan 5 variabel DataFrame yang berbeda-beda menjadi 1 DataFrame agar data-data komentar dari video debat babak pertama hingga babak kelima yang ditayangkan oleh channel YouTube KPU RI tidak terpisah-pisah dan tersentralisasi (terpusat) ke dalam 1 file dataset, saya menggunakan sebuah parameter yaitu ignore_index yang bernilai Boolean yaitu True. Parameter tersebut saya gunakan (ignore_index = True) bertujuan agar indeks pada DataFrame (dalam hal ini indeks yang terdapat di dalam variabel DataFrame "df_comment_debate_video_kpuri") tidak terduplikasi dan tidak mengulangi nilai indeks yang sama dari indeks awal.

Sebagai contoh variabel DataFrame "first_video_debate_df" memiliki sebuah nilai awal indeks yaitu 0 (indeks data 0 mengartikan jika data tersebut merupakan baris pertama yang ada di dalam data tersebut) dan indeks terakhir pada variabel DataFrame "first_video_debate_df" adalah 5399 yang mengartikan bahwa baris data tersebut merupakan baris data ke-5.400. Lalu karena DataFrame "first_video_debate_df" digabungkan dengan DataFrame "second_video_debate_df" maka baris data selanjutnya sudah pasti merupakan data yang dimiliki oleh indeks data ke-0/baris pertama dari DataFrame "second_video_debate_df".

Jika tidak menggunakan parameter ignore_index yang bernilai True, maka variabel DataFrame "df_comment_video_debate_kpuri" yang telah menjadi DataFrame tersentralisasi tidak melanjutkan nilai indeks yang selanjutnya yaitu indeks data ke-5400 yang merupakan data pertama/indeks data ke-0 dari variabel DataFrame "second_video_debate_df". Jadi secara tidak langsung akan ada 2 baris data yang memiliki indeks ke-0 jika DataFrame "df_comment_video_debate_kpuri" berisikan data-data komentar dari DataFrame "first_video_debate_df" dan "second_video_debate_df". Karena DataFrame yang tersentralisasi ini berisikan data-data komentar hasil penggabungan 5 variabel DataFrame yang berbeda, maka jika tidak menggunakan parameter ignore_index bernilai True DataFrame "df_comment_debate_video_kpuri" memiliki 5 baris data yang memiliki indeks data ke-0

Jadi alasan tersebutlah saya menggunakan parameter ignore_index = True untuk menghindari terjadinya duplikasi indeks data pada DataFrame yang tersentralisasi/terpusat yaitu variabel DataFrame "df_comment_debate_video_kpuri".

In [14]:
# Menampilkan 8 data terbawah yang dimiliki oleh variabel DataFrame "df_comment_video_debate_kpuri"
df_comment_video_debate_kpuri.tail(8)

video_id                channel_id                  comment_id  \
12732  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgzhoddgrUPP83dT0t54AaABAg   
12733  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz2hYFmk7PWa7BQoXV4AaABAg   
12734  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgyQ2MRlfYE1vxyBu3p4AaABAg   
12735  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgxHk2YguNKSdcx_Mcp4AaABAg   
12736  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgxV7GJULI9y3HvArNF4AaABAg   
12737  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgzMxktSEwtzZfG2Fa54AaABAg   
12738  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugx_JIr_4N5AC8KJcbV4AaABAg   
12739  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugx4nJkrQUve_UheMdp4AaABAg   

                       author  \
12732              @Lilis-279   
12733     @dudehdewangsih8541   
12734  @sitinadasalsabila9520   
12735  @sitinadasalsabila9520   
12736       @tonykustiadi3213   
12737   @syahrulramadhani2365   
12738        @rasmibaderi1057   
12739            @opanboc5038   

                                            text_comment  like_count  \
12732  KPU yang terhormat tolong awasi dan pantau sel...           0   
12733  Siapapun president nya ketika dia amanah jujur...           0   
12734                                       Amin Aamiin.           0   
12735                                                😮😮😮           0   
12736             TUHAN YME YG MENENTUKAN PEMIMPIN NKRI❤           0   
12737  l😊😊lp<br>p.o..olloo.😊.ok<br>o<br>lloo😊ol.ooo😊o...           0   
12738  Yg lebih penting juga, masalah pendidikan kepe...           0   
12739  Gak ada Keraguan Untuk Pasangan AMIN....Seanda...           0   

       total_reply                                    channel_url  \
12732            0              http://www.youtube.com/@Lilis-279   
12733            0     http://www.youtube.com/@dudehdewangsih8541   
12734            0  http://www.youtube.com/@sitinadasalsabila9520   
12735            0  http://www.youtube.com/@sitinadasalsabila9520   
12736            0       http://www.youtube.com/@tonykustiadi3213   
12737            0   http://www.youtube.com/@syahrulramadhani2365   
12738            0        http://www.youtube.com/@rasmibaderi1057   
12739            0            http://www.youtube.com/@opanboc5038   

           published_date        updated_date  
12732 2024-02-04 22:37:14 2024-02-04 22:37:14  
12733 2024-02-04 22:38:29 2024-02-04 22:38:29  
12734 2024-02-04 22:41:39 2024-02-04 22:41:39  
12735 2024-02-04 22:42:06 2024-02-04 22:42:06  
12736 2024-02-04 22:45:54 2024-02-04 22:45:54  
12737 2024-02-04 22:50:55 2024-02-04 22:50:55  
12738 2024-02-04 23:01:15 2024-02-04 23:01:15  
12739 2024-02-04 18:22:22 2024-02-04 18:22:22

Berdasarkan hasil diatas dapat diketahui jika indeks data ke-12739 atau yang merupakan baris data 12.740 sama dengan data komentar yang dimiliki oleh variabel DataFrame "fifth_video_debate_df" yang merupakan indeks data ke-1799 atau pada baris data 1.800 sehingga proses penggabungan data-data komentar yang didapatkan dari video debat babak pertama hingga babak kelima capres-cawapres Republik Indonesia tahun 2024 telah berhasil dilakukan dan urutan indeks dari variabel DataFrame yang tersentralisasi (df_comment_video_debate_kpuri) sudah sesuai dengan output yang diharapkan.

In [15]:
# Menampilkan jumlah data duplikat yang dimiliki di dalam variabel DataFrame "df_comment_debate_video_kpuri"
df_comment_video_debate_kpuri.duplicated().sum()

0

In [16]:
"""
Menampilkan jumlah nilai data yang hilang (missing values) dari masing-masing
kolom yang terdapat di dalam variabel DataFrame "df_comment_debate_video_kpuri"
"""
df_comment_video_debate_kpuri.isnull().sum()

video_id           0
channel_id         0
comment_id         0
author            10
text_comment       0
like_count         0
total_reply        0
channel_url       10
published_date     0
updated_date       0
dtype: int64

In [17]:
"""
Menampilkan beberapa informasi dasar dari setiap kolom seperti tipe data,
jumlah baris data atau kolom yang dimiliki oleh DataFrame, dll., yang dimiliki
oleh variabel DataFrame "df_comment_video_debate_kpuri"
"""
df_comment_video_debate_kpuri.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12740 entries, 0 to 12739
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   video_id        12740 non-null  object        
 1   channel_id      12740 non-null  object        
 2   comment_id      12740 non-null  object        
 3   author          12730 non-null  object        
 4   text_comment    12740 non-null  object        
 5   like_count      12740 non-null  int64         
 6   total_reply     12740 non-null  int64         
 7   channel_url     12730 non-null  object        
 8   published_date  12740 non-null  datetime64[ns]
 9   updated_date    12740 non-null  datetime64[ns]
dtypes: datetime64[ns](2), int64(2), object(6)
memory usage: 995.4+ KB


Berdasarkan output yang didapatkan dari fungsi "info()" yang menunjukkan beberapa informasi dasar yang terdapat di dalam DataFrame, kita dapat mengetahui beberapa poin informasi dari variabel DataFrame "df_comment_video_debate_kpuri" diantaranya adalah sebagai berikut:

1. Variabel DataFrame "df_comment_video_debate_kpuri" merupakan sebuah objek berupa DataFrame yang berisikan kumpulan data yang didalamnya terdapat baris, kolomda dan tabel (dataset) dari hasil proses penggabungan beberapa data-data komentar yang didapatkan dari video YouTube tentang debat calon Presiden-Wakil Presiden Republik Indonesia tahun 2024 babak pertama hingga  babak kelima yang ditayangkan oleh channel YouTube KPU RI.

2. Variabel DataFrame "df_comment_video_debate_kpuri" memiliki jumlah data/baris sebanyak 12.740 data yang terdiri dari teks komentar yang dikirimkan oleh penonton video tersebut, nama akun yang berkomentar, tanggal dan waktu dari komentar tersebut dibuat, dll. Data-data tersebut didapatkan melalui proses penggabungan kelima variabel DataFrame yang masing-masing memiliki jumlah data/informasi komentar yang berbeda-beda seperti pada variabel DataFrame "first_debate_video_df" memiliki baris data sebanyak 5.400 informasi komentar, variabel DataFrame "second_debate_video_df" memiliki baris data sebanyak 1.900 data komentar, dst.

3. DataFrame "df_comment_video_debate_kpuri" memiliki 10 kolom yang diantaranya adalah kolom "video_id", "channel_id", "comment_id", "author", "text_comment", "like_count", "total_reply", "channel_url", "published_date" dan terakhir adalah kolom "updated_date".

4. Masing-masing kolom yang ada di dalam DataFrame "df_comment_video_debate_kpuri" tidak memiliki nilai NaN (Not a Number) atau nilai yang hilang (missing value) sehingga dapat disimpulkan bahwa sepuluh kolom yang dimiliki oleh variabel DataFrame "df_comment_video_debate_kpuri" dan setiap baris indeksnya dimiliki oleh 10 kolom, tidak ada satupun baris data pada indeks yang tidak memiliki nilai/data.

5. Pada variabel DataFrame "df_comment_video_debate_kpuri" terdapat 3 jenis tipe data yang berbeda. Tipe data yang pertama adalah tipe data Integer untuk setiap kolom yang memiliki nilai data bersifat yang angka atau numerik. Data numerik/data angka dibedakan menjadi 2 jenis bilangan angka yaitu bilangan bulat dan bilangan desimal. Angka bilangan bulat direpresentasikan sebagai tipe data Integer dan angka bilangan desimal direpresentasikan sebagai tipe data Float. Karena pada DataFrame "df_comment_video_debate_kpuri" hanya memiliki 2 kolom saja yang bertipe data Integer, yaitu kolom "like_count" dan "total_reply", di mana kedua kolom tersebut memiliki nilai angka bilangan bulat, maka kolom tersebut bertipe data Integer bukan Float. Tipe data yang kedua adalah tipe data Datetime dan hanya 2 kolom saja yang memiliki tipe data Datetime yaitu kolom "published_date" dan "updated_date". Tipe data yang ketiga yang dimiliki oleh variabel DataFrame "df_comment_video_debate_kpuri" adalah tipe data Object/String. Tipe data Object/String adalah representasi dari nilai-nilai data teks, simbol-simbol maupun karakter spesial yang dimiliki oleh suatu kolom di dalam DataFrame. Secara jumlah, tipe data Object/String pada variabel DataFrame "df_comment_video_debate_kpuri" dimiliki oleh 6 kolom (terbanyak diantara tipe data Integer ataupun Datetime) yaitu kolom "video_id", "channel_id", "comment_id", "author", "text_comment" dan kolom "channel_url".

In [18]:
"""
Mengubah urutan data yang dimiliki oleh variabel DataFrame "df_comment_video_debate_kpuri"
berdasarkan jumlah like terbanyak hingga jumlah like paling sedikit yang didapatkan/diterima oleh suatu teks komentar
diantara 12.740 data komentar yang merupakan kumpulan data hasil penggabungan komentar-komentar dari video
debat babak pertama hingga babak kelima capres-cawapres Republik Indonesia yang ditayangkan oleh channel YouTube KPU RI
"""

df_comment_video_debate_kpuri = df_comment_video_debate_kpuri.sort_values(by=["like_count"],
                                                                          ascending = False, ignore_index = True)

df_comment_video_debate_kpuri.head(38) # Menampilkan 38 data komentar teratas yang memiliki jumlah like terbanyak

video_id                channel_id                  comment_id  \
0   KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugy7bfVbuDxBgae9eVp4AaABAg   
1   8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz8GZ3Lfw7X9kYnqSV4AaABAg   
2   YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  UgyBL7L6U5bXWuuzlap4AaABAg   
3   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgzXkcSL0MuTJL6ZpY54AaABAg   
4   KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgwXBn-0QCB6GntoHY94AaABAg   
5   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgzF-687IdOdv5qeEsp4AaABAg   
6   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxbXCPeUDV_zPUhZjF4AaABAg   
7   KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugw-FFWX0IY1b5sW77x4AaABAg   
8   yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugw_i1DjvR-QcFZOWFR4AaABAg   
9   KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgyLvJxPNZznf4VD_hF4AaABAg   
10  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgwOYSRHpyihNxbUL_l4AaABAg   
11  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgzE0Z69Qxhmo7W7rG54AaABAg   
12  anuQxiXpQ7I  UCVdSz86o9q2cQKvAQ5QhOYg  Ugy8BsMRwY9YzhEKdUN4AaABAg   
13  YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  Ugx-_DJpJiZRtfv7FMF4AaABAg   
14  YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  UgyTWKUniSQGlIiAAdd4AaABAg   
15  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxZddA-eb4ajHIz-vd4AaABAg   
16  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgzZITGXd0oboGAaMEp4AaABAg   
17  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  UgwHox027DBKmVb_9e54AaABAg   
18  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz2mQvTxHONMD982514AaABAg   
19  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxCVWBXBHS2aAExXJ14AaABAg   
20  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugx8Uk2IjomoD16znX94AaABAg   
21  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugxo25xaQwZnNIDfCkl4AaABAg   
22  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugx2150EevOnGTsC6bB4AaABAg   
23  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgwG8FVmd_-g6-XkTVd4AaABAg   
24  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxJgM761WF2PFOlnoJ4AaABAg   
25  8J66JxvmEzo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugy-S-3oeLCUjmfVhUt4AaABAg   
26  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgwiUVXS_YjXcTJ0ULB4AaABAg   
27  YzC828FYrwM  UCVdSz86o9q2cQKvAQ5QhOYg  UgxJyLPfPdq4lY8kmM14AaABAg   
28  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgwBmgAzm9kt0Uu1_Et4AaABAg   
29  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  Ugxm0_t-p32oVgxoVn54AaABAg   
30  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgxEU6Pyh8OU1kV_am54AaABAg   
31  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgxZ7WP3Np2jAOVGaOl4AaABAg   
32  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgyUHi1eyviFsjfdXtl4AaABAg   
33  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgyNp2gFN-BOUgKJlgB4AaABAg   
34  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgwFl1-jiOxcmyWJN-94AaABAg   
35  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  Ugz0_cQ4OgpUqNaLtzh4AaABAg   
36  KJdt-HBBGIo  UCVdSz86o9q2cQKvAQ5QhOYg  UgyHfEX3PkJTnU2J2ot4AaABAg   
37  yNO0YS846kU  UCVdSz86o9q2cQKvAQ5QhOYg  UgxCr7dsBpQMoZwK-A14AaABAg   

                     author  \
0          @nabilbudiman271   
1        @anisluthfiani9056   
2          @nabilbudiman271   
3          @faridafasya9815   
4            @Teman_Bambank   
5            @rizkiindah479   
6          @nabilbudiman271   
7              @waryati5554   
8       @ajipratamaputra815   
9     @angkadandata_channel   
10          @sodrishandy545   
11             @robinong153   
12             @kajikampung   
13    @aniksetianingsih2011   
14           @staypolid5125   
15      @ariesty_crispy2468   
16             @bintang4783   
17                 @sxn1707   
18           @misyebima8609   
19            @viedaily3572   
20         @Nurwahida_Ahmad   
21  @ismailash-shanhaji1053   
22                @isalutfi   
23  @afiyahsetianingrum0158   
24           @indahraya4390   
25      @nurulmutmainna1914   
26       @hendriakhtiri5839   
27               @bangjok22   
28             @herisalmani   
29               @restimita   
30          @vidiokeren6134   
31      @wahyuuusaputra3524   
32           @rialfachrezi8   
33           @dzikrifeature   
34            @muflihrz4074   
35           @diary_nay7617   
36          

## 3. Export DataFrame Variable into Dataset with Spreadsheet Format (Microsoft Excel)

In [19]:
"""
Melakukan proses penyimpanan/meng-eksport data atau informasi komentar utama sebanyak 12.740 dari hasil
penggabungan data-data dari kelima variabel DataFrame yang didapatkan dari hasil web scraping pada
video debat capres-cawapres babak pertama hingga babak kelima yang di-upload oleh Channel YouTube KPU RI
"""

df_comment_video_debate_kpuri.to_excel("../Dataset/Dataset Komentar Video Debat dari YouTube Channel KPU RI.xlsx",
                                             index = False, header = True)